client/plots/violin.interactivity.js

```diff
@@ -143,7 +145,16 @@ export function setInteractivity(self) {
 		for (const [c, k] of Object.entries(data.samples)) {
 			const sampleIdObj = {}
 			if (data.refs.bySampleId[c]) {
-				sampleIdObj[data.refs.bySampleId[c]] = k[term.$id].value
+				if (self.config.settings.violin?.unit === 'log' && k[term.$id].value != 0) {
+					sampleIdObj[data.refs.bySampleId[c]] = roundValue(
+						getBaseLog(
```

Comment:
you can use `Math.log2()`

Reply:
well, I intentionally did this in case they wanted a different log base. It's a lot neater this way. If it changes then we won't have to replace the code everywhere and just update the db file.